DSharpPlus/Entities/Guild/DiscordGuild.cs

```diff
@@ -1069,7 +1069,14 @@ public DiscordRole GetRole(ulong id)
         /// <returns>Requested channel.</returns>
         /// <exception cref="Exceptions.ServerErrorException">Thrown when Discord is unable to process the request.</exception>
         public DiscordChannel GetChannel(ulong id)
-            => (this._channels != null && this._channels.TryGetValue(id, out var channel)) ? channel : null;
+        {
+            // force the cache to update if it is either null or the channel is not found
+            if (!this._channels.TryGetValue(id, out var channel) || this._channels == null)
+                this._channels = (ConcurrentDictionary<ulong, DiscordChannel>)
+                    this.Discord.ApiClient.GetGuildChannelsAsync(this.Id).GetAwaiter().GetResult();
```

Comment:
This method [does not set the client](https://github.com/DSharpPlus/DSharpPlus/blob/master/DSharpPlus/Net/Rest/DiscordApiClient.cs#L1139-L1157)  meaning any REST calls (or even relational properties) will throw an exception

Reply:
Oh no wait I'm blind don't mind me

Reply:
But it doesn't set the guild, so that's still of concern

Reply:
> But it doesn't set the guild, so that's still of concern

Ill look into it later but I think this how GetChannelsAsync does it as well. Because Discord sends the guild id, and the channel object just uses the DiscordClient to get the concrete guild.